### PR TITLE
feat(wallet): Warning for Token Account Creation

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Will now display a `Warning` message for a `SolanaSPLTokenTransferWithAssociatedTokenAccountCreation` transaction.

Resolves <https://github.com/brave/brave-browser/issues/22260>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Send SPL Token With Create Associated Token Account:

- Create a second `Solana` account
- Select the `Send` tab
- Select the `Solana Mainnet Beta` network and your first `Solana` account
- Select a `SPL` token and enter the amount you want to send
- Enter the second `Solana` address to send to
- Click `Send`
- On the `Confirm Transaction` panel there should be a warning message, click `Confirm`
- On the `Accounts` tab, select your `Solana` account and make sure the `Transaction` is visible and wait for a `Confirmed` status.

https://user-images.githubusercontent.com/40611140/163608634-76a8160b-0189-4d2a-91c5-19f7c545176c.mov

Screen Shot 2022-04-14 at 3 37 19 PM

### CODE REVIEW

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx [hunk 1]
```diff
@@ -128,6 +130,14 @@ function ConfirmTransactionPanel ({
     setShowAdvancedTransactionSettings(!showAdvancedTransactionSettings)
   }
 
+  const onClickLearnMore = () => {
+    chrome.tabs.create({ url: 'https://support.brave.com/hc/en-us/articles/5546517853325' }, () => {
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx:134]: Needs link to actual article

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts [hunk 2]
```diff
@@ -252,7 +253,7 @@ export const AssetIcon = AssetIconFactory<AssetIconProps>({
 })
 
 // [FIXME]: Duplicated from components/brave_wallet_ui/components/extension/sign-panel/style.ts
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts:255]: Might be a good time to fix this too

Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/confirm-transaction-panel/style.ts]: Moved to `shared-panel-styles.ts`

File: components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx:134]: This is the actual link, article just hasn't been created yet.

**Review by josheleonard — APPROVED** [review]
LGTM, just needs the "Learn more" article published

**Pull request merged**